src/lib.rs

```diff
@@ -2,23 +2,37 @@
 use pyo3::{prelude::*, wrap_pyfunction};
 
 use cpy_binder::export_cpy;
+use lazy_static::lazy_static;
+use std::sync::Mutex;
 
 struct NavigationManager {
     navigator: navigator_rs::Navigator,
 }
+
+lazy_static! {
+    static ref NAVIGATOR: Mutex<Option<NavigationManager>> = Mutex::new(None);
+}
+
 impl NavigationManager {
     // private constructor to prevent direct instantiation
-    fn new() -> NavigationManager {
-        NavigationManager {
-            navigator: navigator_rs::Navigator::new(),
+    fn new() -> &'static Mutex<Option<Self>> {
+        let mut st = NAVIGATOR.lock().unwrap();
+        if st.is_none() {
+            *st = Some(NavigationManager {
+                navigator: navigator_rs::Navigator::new(),
+            });
+            &NAVIGATOR
+        } else {
+            panic!("NavigationManager is already initialized")
```

Comment:
We should not abort, there is no reason for it, we could just populate NAVIGATOR content if there is nothing there, and if it's already initialized it, ignore.